src/UnitTests/Helpers/TestBaseClass.cs

```diff
@@ -7,14 +8,35 @@
 /// </summary>
 public class TestBaseClass : IEntryExitDecorator
 {
-    public void OnEntry()
+    public virtual void OnEntry()
     {
         // Ensure that every test has the InUnitTestRunner flag
         // set, so threading doesn't go nuts.
         Splat.ModeDetector.Current.SetInUnitTestRunner(true);
     }
 
-    public void OnExit()
+    public virtual void OnExit()
     {
     }
 }
+
+public class TempFileBaseClass : TestBaseClass
+{
+    public DirectoryInfo Directory { get; set; }
+
+    public override void OnEntry()
+    {
+        var f = Path.GetTempFileName();
+        var name = Path.GetFileNameWithoutExtension(f);
+        File.Delete(f);
+        Directory = new DirectoryInfo(Path.Combine(Path.GetTempPath(), name));
+        Directory.Create();
+        base.OnEntry();
+    }
```

Comment:
I'm a little wary of having unit tests that touch the file system. I'm fine with them now if they're fast. It'd be a pain to mock out all the fs operations, although that is what I built https://github.com/haacked/rothko for. :smile:

Or should we add an `IntegrationTests` project and move our tests of git operations there?